spec/controllers/api/canvas_proxy_controller_spec.rb

```diff
@@ -37,6 +37,56 @@
     end
   end
 
+  # This functionality lives in application_controller.rb but we need
+  # a real controller to test it
+  describe "rescue from LMS::Canvas::RefreshTokenRequired" do
+    before do
+      request.headers["Authorization"] = @user_token_header
+      allow(controller.request).to receive(:host).and_return("example.com")
+      def controller.proxy
+        @auth = FactoryGirl.create(:authentication)
+        raise LMS::Canvas::RefreshTokenRequired.new("", nil, @auth)
+      end
+    end
+    context "application instance allows user token" do
+      before do
+        allow(controller).to receive(:current_application_instance).and_return(@application_instance)
+        allow(Application).to receive(:find_by).with(:lti_key).and_return(@application_instance)
+      end
+      it "deletes the authentication and returns a status forbidden" do
+        type = "LIST_ACCOUNTS"
+        get :proxy, params: { lms_proxy_call_type: type, lti_key: @application_instance.lti_key }, format: :json
+        expect(response).to have_http_status(:forbidden)
+        auth = Authentication.find(@auth.id) rescue nil
```

Comment:
You can `auth = Authentication.find_by(id: @auth.id)` and that will return nil if nothing is found.

Reply:
But really, shouldn't there be an auth in your spec?

Reply:
Oh, I see, you want it to be nil